### PR TITLE
Typescript typings

### PR DESCRIPTION
**Pull request by xkr47**
Added typescript bindings and some es6/typescript notes in the readme.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -32,6 +32,12 @@ console.log(results);
 */
 ```
 
+In ES6/Typescript the first line should look like this instead:
+
+```js
+import unquotedValidator = require('unquoted-property-validator');
```
Comment (pgilad) [inline readme.md]: Should be from no?

Comment (xkr47) [reply readme.md]: I couldn't get it to work using the "normal" syntaxes (inluding from) so I used the solution suggested here and it worked: https://stackoverflow.com/questions/39415661/what-does-resolves-to-a-non-module-entity-and-cannot-be-imported-using-this#39415662

Before I added the proposed typescript bindings in my local project I was avble to use it like this:
```es
import * as unquotedValidator from 'unquoted-property-validator';
```

**Comment by xkr47** [comment]
I apologize, I guess the `index.d.ts` should also be added to the `"files"` section in the `package.json` file so that it gets included in the release package. Now it isn't and so the typings don't work.. Hope I didn't make any other mistakes too..

**Comment by xkr47** [comment]
Oh boy the index.d.ts is broken too.. I'll open a new PR for both this and the above mistake too.

**Pull request merged**